script/releases/v0.1-eigenpod-example/1-eoa.s.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "zeus-templates/templates/EOADeployer.sol";
+
+import "src/contracts/pods/EigenPod.sol";
+import "src/contracts/pods/EigenPodManager.sol";
+import "src/contracts/interfaces/IEigenPodManager.sol";
+
+import "zeus-templates/utils/StringUtils.sol";
+
+contract DeployEigenPodAndManager is EOADeployer {
+    using StringUtils for string;
+
+    /// @notice deploys an EigenPod and returns the deployed address
+    function _deploy(Addresses memory, Environment memory, Params memory params) internal override returns (Deployment[] memory) {
+
+        vm.startBroadcast();
+
+        // create a defunct EigenPodManager
+        address newEigenPodManager = address(new EigenPodManager(
+            IETHPOSDeposit(address(1)),
+            IBeacon(address(2)),
+            IStrategyManager(address(3)),
+            ISlasher(address(4)),
+            IDelegationManager(address(5))
+        ));
+
+        // record new deployment
+        _deployments.push(Deployment({
+            name: type(EigenPodManager).name,
+            deployedTo: newEigenPodManager,
+            envToUpdate: "eigenPodManager.pendingImpl"
+        }));
+
+        // create and record new EigenPod pointing to defunct EigenPodManager
+        _deployments.push(Deployment({
+            name: type(EigenPod).name,
+            deployedTo: address(new EigenPod(
+                IETHPOSDeposit(params.ethPOS),
+                IEigenPodManager(newEigenPodManager), // update EigenPodManager address
+                params.EIGENPOD_GENESIS_TIME
+            )),
+            envToUpdate: "eigenPod.pendingImpl"
```

Comment:
Interesting -- this seems like it's leaking a bit of complexity. I figured zeus would take care of this given the `name` field?

Reply:
If zeus isn't able to handle this for us, we should have constants defined or something that makes it harder to supply the wrong value here.

And I guess I'd question whether the `name` field is relevant

Reply:
Yeah, I think a better way to do this is to have that `envToUpdate` field be a boolean, either to update the `name` reserved config value's pending impl or not. Good callout

Reply:
Discussion:
* Unclear whether we're passing deployments between scripts via env variables or as updated values in JSON config. Should clarify with Justin what he expects/has been building - Nadir/I are fine with either.
* Regardless of whether we're using env variables or updating JSON, the script writer should not have to pass in raw strings to reference storage locations.
  * If using env variables, we will need a library mapping strings to env variables
  * If using JSON, we should be able to discern the location in the JSON by referencing the field getting updated (e.g. by passing in `addrs.eigenPod.pendingImpl` and the abstraction resolves this to the correct JSON key)
  
Other observations:
* Moving solely to env variables might be the play eventually - possibly less brittle.
  * Backwards-compatibility issues will probably exist regardless of using JSON or ENV variables, so different versions may require different env/json schemas